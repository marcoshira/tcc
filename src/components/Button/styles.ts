import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    color: ${theme.colors.white};
    background: ${theme.colors.primaryColor};
    border-radius: 7px;
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.ltmedium};
    font-weight: 500;
    margin-bottom: 4px;
    transition: ${theme.transitions.faster};

    &:disabled {
      background: ${theme.colors.grey};
      color: ${theme.colors.white};
      border: 2px solid ${theme.colors.grey};
      transition: none;

      &:hover {
        background: ${theme.colors.grey};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.grey};
        cursor: not-allowed;
      }
    }
    &:hover {
      cursor: pointer;
      background: inherit;
      color: ${theme.colors.primaryColor};
    }

    @media ${theme.media.gteMedium} {
      /* width: 33%; */
    }
  `}
`;
