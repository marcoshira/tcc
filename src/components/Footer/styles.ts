import styled, { css } from 'styled-components';

export const Footer = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primaryColor};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: white;
    padding: ${theme.spacings.small} 0 ${theme.spacings.medium};

    p {
      margin: 0;
      font-size: 20px;
    }
    a {
      font-size: 18px;
      text-decoration: none;
    }

    @media ${theme.media.gteMedium} {
      padding: ${theme.spacings.small} 8% ${theme.spacings.large};
      flex-direction: row;
      justify-content: space-around;
    }
  `}
`;

export const DevContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    margin-bottom: 20px;

    @media ${theme.media.gteMedium} {
      margin-bottom: 0px;
      p {
        margin-bottom: 20px;
        font-size: 25px;
      }
      a {
        font-size: 25px;
      }
    }
  `}
`;

export const IconsContainer = styled.div`
  ${({ theme }) => css`
    margin-top: 15px;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    p {
      text-align: center;
    }
    svg {
      color: white;
      height: 40px;
    }
  `}
`;

export const IconsInnerContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    svg {
      color: white;
      height: 40px;
      margin-right: 10px;
    }
    a {
      text-decoration: none;
    }
  `}
`;
