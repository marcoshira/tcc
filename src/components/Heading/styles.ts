import styled, { css } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.huge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size ${theme.font.sizes.xlarge}
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, size, uppercase }) => css`
    color: white;
    margin: 0;
    margin-bottom: ${theme.spacings.small};
    padding: ${theme.spacings.large};
    background: ${theme.colors.primaryColor};
    text-align: center;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
