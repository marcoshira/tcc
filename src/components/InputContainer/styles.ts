import styled, { css } from 'styled-components';

type BorderRed = {
  red?: boolean;
  yellow?: boolean;
};

const redBorder = () => css`
  box-shadow: 0px 0px 5px red;
`;

const yellowBorder = () => css`
  box-shadow: 0px 0px 5px yellow;
`;

export const Wrapper = styled.div<BorderRed>`
  ${({ theme, red, yellow }) => css`
    box-shadow: 0px 1px 9px ${theme.colors.primaryColor};
    border-radius: 7px;
    margin-bottom: ${theme.spacings.small};
    width: 100%;
    transition: ${theme.transitions.slowest};

    ${red && redBorder()}

    @media ${theme.media.gteMedium} {
      margin-bottom: ${theme.spacings.large};
    }

    > div {
      > svg {
        position: absolute;
        top: 12px;
        right: 12px;
        width: 32px;
        color: ${theme.colors.primaryColor};

        &:hover {
          color: ${theme.colors.orange};
        }
      }
      &:hover {
        & .tooltiptext2 {
          visibility: visible;
          opacity: 1;
        }
      }
    }

    & .tooltiptext2 {
      visibility: hidden;
      width: 120px;
      background-color: transparent;
      color: gray;
      text-align: center;
      padding: 15px;
      border-radius: 6px;
      border: 1px solid gray;
      /* height: 100%; */
      width: ${theme.frameSizes.small};
      top: 0px;
      left: 101.5%;

      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;

      text-align: center;
    }
    & .tooltiptext2::before {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 100%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent gray transparent transparent;
    }
    & .tooltiptext2::after {
      content: ' ';
      position: absolute;
      top: 26px;
      right: 99.6%;
      margin-top: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: transparent white transparent transparent;
    }

    /* Tooltip container */
    &.tooltip {
      position: relative;
      display: inline-block;
      border-bottom: 1px dotted black; /* If you want dots under the hoverable text */
    }

    /* Tooltip text */
    &.tooltip .tooltiptext {
      visibility: hidden;
      width: 120px;
      background-color: red;
      color: white;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid red;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 2;

      opacity: 0;
      transition: opacity 0.3s;
    }

    &.tooltip .tooltiptext4 {
      visibility: hidden;
      width: 120px;
      background-color: #fff3ac;
      color: #856404;
      text-align: center;
      padding: 5px 0;
      border-radius: 6px;
      border: 2px solid #fff3ac;
      height: ${theme.spacings.large};
      width: 100%;
      bottom: 106%;
      left: 50%;
      margin-left: -50%; /* Use half of the width (120/2 = 60), to center the tooltip */

      /* Position the tooltip text - see examples below! */
      position: absolute;
      z-index: 1;

      opacity: 0;
      transition: opacity 0.3s;
    }

    /* Show the tooltip text when you mouse over the tooltip container */
    &.tooltip:hover .tooltiptext,
    &.tooltip:hover .tooltiptext4 {
      visibility: visible;
      opacity: 1;
    }

    &.tooltip .tooltiptext::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -9px;
      border-width: 9px;
      border-style: solid;
      border-color: red transparent transparent transparent;
    }

    &.tooltip .tooltiptext4::after {
      content: ' ';
      position: absolute;
      top: 100%; /* At the bottom of the tooltip */
      left: 50%;
      margin-left: -10px;
      border-width: 10px;
      border-style: solid;
      border-color: #fff3cd transparent transparent transparent;
    }
  `}
`;
