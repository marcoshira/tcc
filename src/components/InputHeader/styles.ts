import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0px 0px;
    display: flex;
    justify-content: center;

    h2 {
      font-size: ${theme.font.sizes.medium};
      font-weight: 300;
    }
  `}
`;
