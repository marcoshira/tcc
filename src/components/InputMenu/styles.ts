import styled, { css } from 'styled-components';
import { InputMenuProps } from '.';

const frameSize = {
  small: (theme) => css`
    height: ${theme.frameSizes.ltxsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  mediumSmall: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.xxsmall};
    }
  `,
  medium: (theme) => css`
    height: ${theme.frameSizes.small};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.ltxsmall};
    }
  `,
  big: (theme) => css`
    height: ${theme.frameSizes.gtsmall};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.small};
    }
  `,
  huge: (theme) => css`
    height: ${theme.frameSizes.smallMedium};
    @media ${theme.media.gteMedium} {
      height: ${theme.frameSizes.gtsmall};
    }
  `,

  spt: (theme, depth) => css`
    height: ${depth < 6
      ? theme.frameSizes.xsmall
      : depth < 8
      ? theme.frameSizes.ltsmall
      : depth < 10
      ? theme.frameSizes.small
      : `${(depth - 10) * 2.5 + 26}rem`};
    @media ${theme.media.gteMedium} {
      height: ${`${(Math.floor(depth / 4) + 1) * 6}rem`};
    }
  `,
};

export const Wrapper = styled.div<InputMenuProps>`
  ${({ theme, size, depth }) => css`
    display: flex;
    justify-content: center;
    height: 0px;
    opacity: 0;

    transition: ${theme.transitions.faster};

    > div {
      display: none;
    }

    &.open {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }

    &.zero {
      height: 0px;
      opacity: 0;
    }

    @media ${theme.media.gteMedium} {
      ${frameSize[size](theme, depth)};
      opacity: 1;

      > div {
        display: flex;
      }
    }
  `}
`;
