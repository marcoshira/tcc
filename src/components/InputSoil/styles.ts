import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 80%;
    flex-direction: column;
    justify-content: center;
  `}
`;

export const TypeContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;

    @media ${theme.media.gteMedium} {
      justify-content: space-around;
    }
  `}
`;

export const Type = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.primaryColor};
    font-size: ${theme.font.sizes.ltmedium};

    transition: ${theme.transitions.fast};

    &.selected {
      color: ${theme.colors.yellow};
    }

    &:hover {
      color: ${theme.colors.orange};
      cursor: pointer;
    }
  `}
`;

export const SubTypeContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    .control {
      display: flex;
      justify-content: end;
      position: relative;
      padding-left: 20px;
      margin-bottom: 5px;
      padding-top: 3px;
      cursor: pointer;
      font-size: ${theme.font.sizes.small};
    }
    .control input {
      position: absolute;
      z-index: -1;
      opacity: 0;
    }
    .control_indicator {
      position: absolute;
      top: 8px;
      left: 0px;
      height: 10px;
      width: 10px;
      background: #e6e6e7;
      border: 0px solid #000000;
    }
    .control:hover input ~ .control_indicator,
    .control input:focus ~ .control_indicator {
      background: #cccccc;
    }

    .control input:checked ~ .control_indicator {
      background: #0a1128;
    }
    .control:hover input:not([disabled]):checked ~ .control_indicator,
    .control input:checked:focus ~ .control_indicator {
      background: #0a1128;
    }
    .control input:disabled ~ .control_indicator {
      background: #e6e6e6;
      opacity: 0.6;
      pointer-events: none;
    }
    .control_indicator:after {
      box-sizing: unset;
      content: '';
      position: absolute;
      display: none;
    }
    .control input:checked ~ .control_indicator:after {
      display: block;
    }
    .control-radio .control_indicator {
      border-radius: 50%;
    }

    .control-radio .control_indicator:after {
      left: 7px;
      top: 7px;
      height: 0px;
      width: 0px;
      border-radius: 50%;
      background: #0a1128;
      transition: background 250ms;
    }
    .control-radio input:disabled ~ .control_indicator:after {
      background: #7b7b7b;
    }

    @media ${theme.media.gteMedium} {
      .control-group {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
      }

      .control {
        display: flex;
        justify-content: end;
        position: relative;

        margin-bottom: 5px;
        padding-top: 3px;
        cursor: pointer;
        font-size: ${theme.font.sizes.small};
      }
    }
  `}
`;
