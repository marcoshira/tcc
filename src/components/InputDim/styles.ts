import styled, { css } from 'styled-components';
import { Wrapper as NumberInput } from '../NumberInput/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large} 0 ${theme.spacings.medium};
    margin: ${theme.spacings.xsmall} 0 ${theme.spacings.medium};
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
  `}
`;

export const NumberContainer = styled.div`
  ${({ theme }) => css`
    width: 100%;

    ${NumberInput} {
      width: 100%;
      .unit {
        left: 98%;
        top: 10px;
        width: auto;
      }
      span {
        width: 60%;
      }
      input {
        width: 23%;
        height: 23px;
        margin-top: 10px;
      }
    }
    @media ${theme.media.gteMedium} {
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      width: 100%;

      ${NumberInput} {
        width: 50%;
        .unit {
          left: 94%;
          top: 0px;
          width: auto;
        }
        span {
          width: 70%;
          font-size: ${theme.font.sizes.ltmedium};
        }
        input {
          width: 23%;
          height: 23px;
          margin-top: 0px;
        }
      }
    }
  `}
`;

export const RadioContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    width: 100%;

    .control-group {
      display: flex;
      flex-direction: row;
      justify-content: space-around;
      width: 100%;
    }

    .control {
      display: flex;
      justify-content: end;
      position: relative;
      padding-left: 20px;
      margin-bottom: 5px;
      padding-top: 3px;
      cursor: pointer;
      font-size: ${theme.font.sizes.small};
    }
    .control input {
      position: absolute;
      z-index: -1;
      opacity: 0;
    }
    .control_indicator {
      position: absolute;
      top: 8px;
      left: 0px;
      height: 10px;
      width: 10px;
      background: #e6e6e7;
      border: 0px solid #000000;
    }
    .control:hover input ~ .control_indicator,
    .control input:focus ~ .control_indicator {
      background: #cccccc;
    }

    .control input:checked ~ .control_indicator {
      background: #0a1128;
    }
    .control:hover input:not([disabled]):checked ~ .control_indicator,
    .control input:checked:focus ~ .control_indicator {
      background: #0a1128;
    }
    .control input:disabled ~ .control_indicator {
      background: #e6e6e6;
      opacity: 0.6;
      pointer-events: none;
    }
    .control_indicator:after {
      box-sizing: unset;
      content: '';
      position: absolute;
      display: none;
    }
    .control input:checked ~ .control_indicator:after {
      display: block;
    }
    .control-radio .control_indicator {
      border-radius: 50%;
    }

    .control-radio .control_indicator:after {
      left: 7px;
      top: 7px;
      height: 0px;
      width: 0px;
      border-radius: 50%;
      background: #0a1128;
      transition: background 250ms;
    }
    .control-radio input:disabled ~ .control_indicator:after {
      background: #7b7b7b;
    }

    span {
      display: none;
    }

    @media ${theme.media.gteMedium} {
      justify-content: space-evenly;
      width: 80%;
      margin-bottom: ${theme.spacings.small};
      span {
        display: inline;
        font-size: ${theme.font.sizes.ltmedium};
      }

      .control-group {
        width: 60%;
      }

      .control {
        font-size: ${theme.font.sizes.gtsmall};
      }
    }
  `}
`;
