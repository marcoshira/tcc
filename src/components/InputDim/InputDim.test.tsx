import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { InputDim, InputDimProps } from '.';

const func = jest.fn();
const func2 = jest.fn();
const func3 = jest.fn();

const props: InputDimProps = {
  onFormChangeArea: func,
  onFormChangeShape: func2,
  onFormChangeComp: func3,
};

describe('<InputDim />', () => {
  it('should render and match snapshot', () => {
    const { container } = renderTheme(<InputDim />);
    expect(
      screen.getByRole('spinbutton', { name: 'Profundidade do Fuste m' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should change input', () => {
    renderTheme(<InputDim {...props} />);
    const inputArea = screen.getByRole('spinbutton', {
      name: 'Diâmetro / Lado da Estaca m',
    });
    const inputLength = screen.getByRole('spinbutton', {
      name: 'Profundidade do Fuste m',
    });
    const inputShapeC = screen.getByRole('radio', {
      name: 'Circunferência',
    });
    const inputShapeQ = screen.getByRole('radio', {
      name: 'Quadrado',
    });

    fireEvent.change(inputArea, { target: { value: 123 } });
    fireEvent.change(inputLength, { target: { value: 125 } });
    fireEvent.click(inputShapeC);
    fireEvent.click(inputShapeQ);

    expect(inputArea).toHaveValue(123);
    expect(inputLength).toHaveValue(125);

    expect(func).toHaveBeenCalled();
    expect(func2).toHaveBeenCalledWith(1);
    expect(func2).toHaveBeenCalledWith(2);
    expect(func3).toHaveBeenCalled();
  });
});
