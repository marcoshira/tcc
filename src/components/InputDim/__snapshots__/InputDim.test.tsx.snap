// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputDim /> should render and match snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 2.4rem;
  position: relative;
}

.c3 span {
  font-size: 1.8rem;
  text-align: center;
}

.c3 .unit {
  color: gray;
  position: absolute;
  left: 97.5%;
  text-align: right;
}

.c3 input {
  border: none;
  border-bottom: 1px solid black;
  width: 17%;
  font-size: 1.8rem;
  text-align: right;
  margin-right: 10px;
}

.c3 input::-webkit-outer-spin-button,
.c3 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c3 input[type='number'] {
  -moz-appearance: textfield;
}

.c3.even input {
  width: 55%;
}

.c0 {
  padding: 0 3.2rem 0 2.4rem;
  margin: 0.8rem 0 2.4rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 100%;
}

.c1 .c2 {
  width: 100%;
}

.c1 .c2 .unit {
  left: 98%;
  top: 10px;
  width: auto;
}

.c1 .c2 span {
  width: 60%;
}

.c1 .c2 input {
  width: 23%;
  height: 23px;
  margin-top: 10px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

.c4 .control-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
}

.c4 .control {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: end;
  -ms-flex-pack: end;
  justify-content: end;
  position: relative;
  padding-left: 20px;
  margin-bottom: 5px;
  padding-top: 3px;
  cursor: pointer;
  font-size: 1.6rem;
}

.c4 .control input {
  position: absolute;
  z-index: -1;
  opacity: 0;
}

.c4 .control_indicator {
  position: absolute;
  top: 8px;
  left: 0px;
  height: 10px;
  width: 10px;
  background: #e6e6e7;
  border: 0px solid #000000;
}

.c4 .control:hover input ~ .control_indicator,
.c4 .control input:focus ~ .control_indicator {
  background: #cccccc;
}

.c4 .control input:checked ~ .control_indicator {
  background: #0a1128;
}

.c4 .control:hover input:not([disabled]):checked ~ .control_indicator,
.c4 .control input:checked:focus ~ .control_indicator {
  background: #0a1128;
}

.c4 .control input:disabled ~ .control_indicator {
  background: #e6e6e6;
  opacity: 0.6;
  pointer-events: none;
}

.c4 .control_indicator:after {
  box-sizing: unset;
  content: '';
  position: absolute;
  display: none;
}

.c4 .control input:checked ~ .control_indicator:after {
  display: block;
}

.c4 .control-radio .control_indicator {
  border-radius: 50%;
}

.c4 .control-radio .control_indicator:after {
  left: 7px;
  top: 7px;
  height: 0px;
  width: 0px;
  border-radius: 50%;
  background: #0a1128;
  -webkit-transition: background 250ms;
  transition: background 250ms;
}

.c4 .control-radio input:disabled ~ .control_indicator:after {
  background: #7b7b7b;
}

.c4 span {
  display: none;
}

@media (min-width:768px) {
  .c3 {
    padding-right: 18px;
    width: 33%;
  }
}

@media (min-width:768px) {
  .c3 .unit {
    left: 90%;
  }
}

@media (min-width:768px) {
  .c3 input {
    width: 6.4rem;
  }
}

@media (min-width:768px) {
  .c3 input::-webkit-outer-spin-button,
  .c3 input::-webkit-inner-spin-button {
    margin-right: 5px;
  }
}

@media (min-width:768px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    width: 100%;
  }

  .c1 .c2 {
    width: 50%;
  }

  .c1 .c2 .unit {
    left: 94%;
    top: 0px;
    width: auto;
  }

  .c1 .c2 span {
    width: 70%;
    font-size: 2rem;
  }

  .c1 .c2 input {
    width: 23%;
    height: 23px;
    margin-top: 0px;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-box-pack: space-evenly;
    -webkit-justify-content: space-evenly;
    -ms-flex-pack: space-evenly;
    justify-content: space-evenly;
    width: 80%;
    margin-bottom: 1.6rem;
  }

  .c4 span {
    display: inline;
    font-size: 2rem;
  }

  .c4 .control-group {
    width: 60%;
  }

  .c4 .control {
    font-size: 1.8rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2 c3"
      >
        <span>
          Diâmetro / Lado da Estaca
        </span>
        <input
          name="price"
          pattern="[0-9]+([.][0-9]+)?"
          placeholder="0.00"
          step="0.01"
          title="This should be a number with up to 2 decimal places"
          type="number"
          value=""
        />
        <span
          class="unit"
        >
          m
        </span>
      </label>
      <label
        class="c2 c3"
      >
        <span>
          Profundidade do Fuste
        </span>
        <input
          name="price"
          pattern="[0-9]+([.][0-9]+)?"
          placeholder="0.00"
          step="0.01"
          title="This should be a number with up to 2 decimal places"
          type="number"
          value=""
        />
        <span
          class="unit"
        >
          m
        </span>
      </label>
    </div>
    <div
      class="c4"
    >
      <span>
        Formato da Estaca:
      </span>
      <div
        class="control-group"
      >
        <label
          class="control control-radio"
        >
          Circunferência
          <input
            checked=""
            name="shape"
            type="radio"
            value="circ"
          />
          <div
            class="control_indicator"
          />
        </label>
        <label
          class="control control-radio"
        >
          Quadrado
          <input
            name="shape"
            type="radio"
            value="quad"
          />
          <div
            class="control_indicator"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
